lib/seed/Seeder.js

```diff
@@ -41,22 +45,34 @@ class Seeder {
   async _listAll(config) {
     this.config = this.setConfig(config);
     const loadExtensions = this.config.loadExtensions;
-    return readdir(this._absoluteConfigDir()).then((seeds) =>
-      seeds
-        .filter((value) => {
-          const extension = path.extname(value);
-          return includes(loadExtensions, extension);
-        })
-        .sort()
+    const seeds = flatten(
+      await Promise.all(
+        this._absoluteConfigDirs().map(getFilepathsInFolderRecursively)
+      )
     );
+    // if true, each dir are already sorted
+    // (getFilepathsInFolderRecursively does this)
+    // if false, we need to sort all the seeds
+    if (this.config.sortDirsSeparately) {
+      return seeds.filter(this._filterByLoadExtensions(loadExtensions));
+    } else {
+      return seeds.filter(this._filterByLoadExtensions(loadExtensions)).sort();
+    }
+  }
+
+  _filterByLoadExtensions(extensions) {
+    return function (value) {
```

Comment:
can be an arrow function for brevity, `(value) => {<...>}`

should this be a class method, though, considering that it is completely static?

Reply:
Where do you put this method in this case, in which file ?

Reply:
can be in this same file for now

Reply:
Done